.circleci/initjob/30-set-collector-version.sh

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env bash
+set -eo pipefail
+
+shared_env=$1
+dockerized_cache_tag=$2
+build_tag=$3
+TAG=$4
+BRANCH=$5
+SOURCE_ROOT=$6
+
+check_use_collector_builder_cache() {
+    use_cache="true"
+
+    if [[ -n "$TAG" ]]; then
+         use_cache="false"
+    fi
+
+    if [[ "$BRANCH" == "master" ]]; then
+         use_cache="false"
+    fi
+
+    if [[ -f pr-metadata/labels/build-builder-image ]]; then
+         use_cache="false"
+    fi
+
+    if [[ -f pr-metadata/labels/valgrind-unit-tests ]]; then
+         use_cache="false"
+    fi
+
+    if [[ -f pr-metadata/labels/valgrind-integration-tests ]]; then
+         use_cache="false"
+    fi
+
+    echo "${use_cache}"
+}
+
+COLLECTOR_VERSION="$(make -s -C "${SOURCE_ROOT}" tag)"
+COLLECTOR_TAG="${COLLECTOR_VERSION}"
+GCLOUD_SSH_FINGERPRINT="$(echo "${GCLOUD_SSH_KEY_PUB}" | awk '{print $2}' | base64 -d | md5sum | awk '{print $1}')"
+GCP_SSH_KEY_FILE="${HOME}/.ssh/id_rsa_${GCLOUD_SSH_FINGERPRINT}"
+
+cat >> "$shared_env" <<- EOF
```

Comment:
In CircleCI << denotes the beginning of a CircleCI parameter. It is escaped so that CircleCI does not interpret it as a parameter. The escape does not work in the shell script so it was removed.